py/private/py_executable.bzl

```diff
@@ -87,7 +87,7 @@ determine_main = rule(
     implementation = _determine_main_impl,
     attrs = {
         "target_name": attr.string(mandatory = True, doc = "The name of the py_binary or py_test we are finding a main for"),
-        "main": attr.string(doc = "Hint the user supplied as the main"),
+        "main": attr.label(doc = "Hint the user supplied as the main", allow_single_file = True),
```

Comment:
This is technically a breaking change as Alex pointed out, so i am not sure if we can land it as a minor.